Mathlib/Order/Interval/Finset/Fin.lean

```diff
@@ -32,37 +43,97 @@ instance instLocallyFiniteOrderTop : ∀ n, LocallyFiniteOrderTop (Fin n)
   | 0 => IsEmpty.toLocallyFiniteOrderTop
   | _ + 1 => inferInstance
 
-variable {n} (a b : Fin n)
+variable {n}
+variable {m : ℕ} (a b : Fin n)
 
-theorem Icc_eq_finset_subtype : Icc a b = (Icc (a : ℕ) b).fin n :=
+@[simp]
+theorem attachFin_Icc :
+    attachFin (Icc a b) (fun _x hx ↦ (mem_Icc.mp hx).2.trans_lt b.2) = Icc a b :=
   rfl
 
-theorem Ico_eq_finset_subtype : Ico a b = (Ico (a : ℕ) b).fin n :=
+@[simp]
+theorem attachFin_Ico :
+    attachFin (Ico a b) (fun _x hx ↦ (mem_Ico.mp hx).2.trans b.2) = Ico a b :=
   rfl
 
-theorem Ioc_eq_finset_subtype : Ioc a b = (Ioc (a : ℕ) b).fin n :=
+@[simp]
+theorem attachFin_Ioc :
+    attachFin (Ioc a b) (fun _x hx ↦ (mem_Ioc.mp hx).2.trans_lt b.2) = Ioc a b :=
   rfl
 
-theorem Ioo_eq_finset_subtype : Ioo a b = (Ioo (a : ℕ) b).fin n :=
+@[simp]
+theorem attachFin_Ioo :
+    attachFin (Ioo a b) (fun _x hx ↦ (mem_Ioo.mp hx).2.trans b.2) = Ioo a b :=
   rfl
 
-theorem uIcc_eq_finset_subtype : uIcc a b = (uIcc (a : ℕ) b).fin n := rfl
+@[simp]
+theorem attachFin_uIcc :
+    attachFin (uIcc a b) (fun _x hx ↦ (mem_Icc.mp hx).2.trans_lt (max a b).2) = uIcc a b :=
+  rfl
 
 @[simp]
-theorem map_valEmbedding_Icc : (Icc a b).map Fin.valEmbedding = Icc ↑a ↑b := by
-  simp [Icc_eq_finset_subtype, Finset.fin, Finset.map_map, Icc_filter_lt_of_lt_right]
+theorem attachFin_Iic : attachFin (Iic a) (fun _x hx ↦ (mem_Iic.mp hx).trans_lt a.2) = Iic a := by
+  ext; simp
 
 @[simp]
-theorem map_valEmbedding_Ico : (Ico a b).map Fin.valEmbedding = Ico ↑a ↑b := by
-  simp [Ico_eq_finset_subtype, Finset.fin, Finset.map_map]
+theorem attachFin_Ico_size : attachFin (Ico a n) (fun _x hx ↦ (mem_Ico.mp hx).2) = Ici a := by
```

Comment:
I find the `_size` quite confusing. I don't think we use this naming convention anywhere else. Maybe change to just
```suggestion
theorem attachFin_Ico_eq_Ici : attachFin (Ico a n) (fun _x hx ↦ (mem_Ico.mp hx).2) = Ici a := by
```